src/messages.cpp

```diff
@@ -881,8 +887,16 @@ namespace quicr::messages {
         buffer << UintVar(msg.group_id);
         buffer << UintVar(msg.object_id);
         buffer.push_back(msg.priority);
-        PushExtensions(buffer, msg.extensions);
-        buffer << UintVar(msg.payload.size());
+        if (!msg.payload.size()) {
+            // empty payload needs a object status to be set
+            buffer << UintVar(msg.payload.size());
+            buffer << UintVar(static_cast<uint8_t>(msg.object_status));
+            PushExtensions(buffer, msg.extensions);
+        } else {
+            buffer << UintVar(msg.payload.size());
+            PushExtensions(buffer, msg.extensions);
+            buffer << msg.payload;
+        }
```

Comment:
Payload size is going to be written either way:
```suggestion
        buffer << UintVar(msg.payload.size());
        if (!msg.payload.size()) {
            // empty payload needs a object status to be set
            buffer << UintVar(static_cast<uint8_t>(msg.object_status));
            PushExtensions(buffer, msg.extensions);
        } else {
            PushExtensions(buffer, msg.extensions);
            buffer << msg.payload;
        }
```

Reply:
Also, prefer using `std::vector::empty` instead of coercing size_t to bool:
```cpp
if (msg.payload.empty()) {

Reply:
Something else just occurred to me; Why bother checking the payload length, if the status flag is already set to "Does not Exist"? We could just check the status. And to that end, if we set "Does not Exist", why even bother checking the payload at all? Could just ignore it entirely (even if there is stuff in it for some weird reason). Should that status be set here instead, to help reduce us manually setting the flag? If the payload is empty then it follows that the status would be set to "Does not exist".